src/components/tests/SourcesTree.spec.js

```diff
@@ -6,32 +6,617 @@ import React from "react";
 import { shallow } from "enzyme";
 import SourcesTree from "../../components/PrimaryPanes/SourcesTree";
 import * as I from "immutable";
+import { showMenu } from "devtools-contextmenu";
+import { copyToTheClipboard } from "../../utils/clipboard";
+import {
+  getShownSource,
+  getSelectedSource,
+  getDebuggeeUrl,
+  getExpandedState,
+  getProjectDirectoryRoot,
+  getSources
+} from "../../selectors";
+
+jest.mock("devtools-contextmenu", () => ({ showMenu: jest.fn() }));
+jest.mock("../../utils/clipboard", () => ({ copyToTheClipboard: jest.fn() }));
+jest.mock("../../selectors");
+
+const defaultSources = I.Map({
+  "server1.conn13.child1/39": createMockSource(
+    "server1.conn13.child1/39",
+    "http://mdn.com/one.js"
+  ),
+  "server1.conn13.child1/40": createMockSource(
+    "server1.conn13.child1/40",
+    "http://mdn.com/two.js"
+  ),
+  "server1.conn13.child1/41": createMockSource(
+    "server1.conn13.child1/41",
+    "http://mdn.com/three.js"
+  )
+});
+
+const singleMockSource = I.Map({
+  "server1.conn13.child1/41": createMockSource(
+    "server1.conn13.child1/41",
+    "http://mdn.com/three.js"
+  )
+});
+
+const singleMockItem = createMockItem(
+  "http://mdn.com/one.js",
+  "one.js",
+  I.Map({ id: "server1.conn13.child1/39" })
+);
+
+const blackboxedMockItem = createMockItem(
+  "http://mdn.com/blackboxed.js",
+  "blackboxed.js",
+  I.Map({ id: "server1.conn13.child1/59" })
+);
+
+const blackboxedMockSource = I.Map({
+  "server1.conn13.child1/59": createMockSource(
+    "server1.conn13.child1/59",
+    "http://mdn.com/blackboxed.js",
+    true
+  )
+});
+
+const mockRootDirectory = {
+  contents: [],
+  name: "root",
+  path: "root/"
+};
+
+const mockDirectory = {
+  contents: [],
+  name: "folder",
+  path: "folder/"
+};
+
+const mockEvent = {
+  preventDefault: jest.fn(),
+  stopPropagation: jest.fn()
+};
+
+let defaultComponent;
```

Comment:
it looks like there is a misunderstanding -- colocate would be to define the variables in the test. for example
```javascript
it('asserts condition 1', () => {
  const { component, props } = render()
  const newProps = { ...props, ...modifiedProps };
  // ... assertion
});

it('asserts condition  2', () => {
  const { component, props } = render()
  // ... assertion
});
```
This current change is making everything global and is quite dangerous. One can easily miss a step and be using the wrong version of a variable that was defined in another test. the test will pass, however there is a possibility of testing the wrong thing. Generally tests should be as isolated from one another as possible. running them in random order should not have any failures.

It is generally better to have something defined close to where you use it. Can you change this either back to what you had or to what I am suggesting?

Reply:
oh ya, I just assumed co-locate meant put everything in the same place. 

Oh so you want the render function to handle the different cases.